src/cli/cmd/mod.rs

```diff
@@ -367,6 +382,113 @@ macro_rules! flakehub_url {
     }};
 }
 
+#[macro_export]
+macro_rules! path {
+    ($root:expr, $($segment:expr),+ $(,)?) => {{
+        let mut path = PathBuf::from($root);
+
+        $(
+            path.push($segment);
+        )+
+
+        path
+    }};
+}
+
+async fn nix_command(args: &[&str], sudo_if_necessary: bool) -> Result<(), FhError> {
+    command_exists("nix")?;
+
+    let is_root_user = nix::unistd::getuid().is_root();
+    let use_sudo = sudo_if_necessary && !is_root_user;
+
+    if use_sudo {
+        tracing::warn!(
+            "Current user is {} rather than root; running Nix command using sudo",
+            whoami::username()
+        );
+        tracing::debug!(
+            "Running: sudo nix --extra-experimental-features 'nix-command-flakes' {}",
+            args.join(" ")
+        );
+    } else {
+        tracing::debug!(
+            "Running: nix --extra-experimental-features 'nix-command-flakes' {}",
+            args.join(" ")
+        );
+    }
+
+    let output = if use_sudo {
+        tokio::process::Command::new("sudo")
+            .args(["nix", "--extra-experimental-features", "nix-command flakes"])
+            .args(args)
+            .stdout(Stdio::inherit())
+            .stderr(Stdio::inherit())
+            .spawn()
+            .wrap_err("failed to spawn Nix command")?
+            .wait_with_output()
+            .await
+            .wrap_err("failed to wait for Nix command output")?
+    } else {
+        tokio::process::Command::new("nix")
+            .args(["--extra-experimental-features", "nix-command flakes"])
+            .args(args)
+            .stdout(Stdio::inherit())
+            .stderr(Stdio::inherit())
+            .spawn()
+            .wrap_err("failed to spawn Nix command")?
+            .wait_with_output()
+            .await
+            .wrap_err("failed to wait for Nix command output")?
+    };
```

Comment:
If you'd like, this can be deduped like this:

```rust
    let mut cmd = if use_sudo {
        let mut cmd = tokio::process::Command::new("sudo");
        cmd.arg("nix");
        cmd
    } else {
        tokio::process::Command::new("nix")
    };
    
    cmd.args(["--extra-experimental-features", "nix-command flakes"]);
    cmd.args(args);
    cmd.stdout(Stdio::inherit());
    cmd.stderr(Stdio::inherit());

    let output = cmd
            .spawn()
            .wrap_err("failed to spawn Nix command")?
            .wait_with_output()
            .await
            .wrap_err("failed to wait for Nix command output")?;
```

Reply:
Good call